openpype/hosts/maya/plugins/publish/extract_look.py

```diff
@@ -133,6 +89,307 @@ def no_workspace_dir():
+class TextureProcessor:
+    def __init__(self, log=None):
+        if log is None:
+            log = logging.getLogger(self.__class__.__name__)
+        self.log = log
+
+    def apply_settings(self, system_settings, project_settings):
+        """Apply OpenPype system/project settings to the TextureProcessor
+
+        Args:
+            system_settings (dict): OpenPype system settings
+            project_settings (dict): OpenPype project settings
+
+        Returns:
+            None
+
+        """
+        pass
+
+    @abstractmethod
+    def process(self,
+                source,
+                colorspace,
+                color_management,
+                staging_dir):
+        """Process the `source` texture.
+
+        Must be implemented on inherited class.
+
+        This must always return a TextureResult even when it does not generate
+        a texture. If it doesn't generate a texture then it should return a
+        TextureResult using the input path and colorspace.
+
+        Args:
+            source (str): Path to source file.
+            colorspace (str): Colorspace of the source file.
+            color_management (dict): Maya Color management data from
+                `lib.get_color_management_preferences`
+            staging_dir (str): Output directory to write to.
+
+        Returns:
+            TextureResult: The resulting texture information.
+
+        """
+        pass
+
+    @staticmethod
+    def get_extension():
+        pass
+
+    def __repr__(self):
+        # Log instance as class name
+        return self.__class__.__name__
+
+
+class MakeRSTexBin(TextureProcessor):
+    """Make `.rstexbin` using `redshiftTextureProcessor`"""
+
+    def process(self,
+                source,
+                colorspace,
+                color_management,
+                staging_dir):
+
+        texture_processor_path = self.get_redshift_tool(
+            "redshiftTextureProcessor"
+        )
+        if not texture_processor_path:
+            raise KnownPublishError("Must have Redshift available.")
+
+        subprocess_args = [
+            texture_processor_path,
+            source
+        ]
+
+        hash_args = ["rstex"]
+        texture_hash = source_hash(source, *hash_args)
+
+        # Redshift stores the output texture next to the input but with
+        # the extension replaced to `.rstexbin`
+        basename, ext = os.path.splitext(source)
+        destination = "{}{}".format(basename, self.get_extension())
+
+        self.log.debug(" ".join(subprocess_args))
+        try:
+            run_subprocess(subprocess_args)
+        except Exception:
+            self.log.error("Texture .rstexbin conversion failed",
+                           exc_info=True)
+            raise
+
+        return TextureResult(
+            path=destination,
+            file_hash=texture_hash,
+            colorspace=colorspace,
+            transfer_mode=COPY
+        )
+
+    @staticmethod
+    def get_extension():
```

Comment:
how about making it a class variable `MakeRSTexBin.extension` instead? I generally try to avoid staticmethod https://google.github.io/styleguide/pyguide.html#217-function-and-method-decorators